internal/providers/conjur/provider.go

```diff
@@ -97,10 +95,17 @@ func ProviderFactory(options plugin_v1.ProviderOptions) (plugin_v1.Provider, err
 		log.Printf("Info: Conjur provider using Kubernetes authenticator-based authentication")
 
 		// Load the authenticator with the config from the environment, and log in to Conjur
-		if authenticator, err = loadAuthenticator(provider.AuthnURL, provider.Version, provider.Config); err != nil {
+		conjurAuthenticatorConf, err = authnConfig.NewConfigFromEnv()
```

Comment:
Since you're making changes in this section of the code...  This may be a personal preference sort of thing, but instead of using a but instead of bunch of `if`...`if-else`...`if-else` statements, we could use a switch:
```
switch {
case provider.Username != "" && provider.APIKey != "":
    // Conjur provider using API key. Do stuff
case tokenFile != "":
    // Conjur provider using access token. Do stuff
case provider.AuthnURL != "" && strings.Contains(provider.AuthnURL, "authn-k8s"):
    // Conjur provider using authenticator. Do stuff
default:
    // Return error
}
```
For me, the switch is easier to see what's going on, but your call, don't want to hold up on this.

Reply:
Looking at the Effective Go [style guide](https://go.dev/doc/effective_go#switch):
> It's therefore possible—**and idiomatic**—to write an if-else-if-else chain as a switch
So I'll make this change

Reply:
Nice, it's much easier to read!